src/vsr.zig

```diff
@@ -815,18 +815,22 @@ test "exponential_backoff_with_jitter" {
 /// This does require that the user specify the same order to all replicas.
 /// The caller owns the memory of the returned slice of addresses.
 pub fn parse_addresses(allocator: std.mem.Allocator, raw: []const u8, address_limit: usize) ![]std.net.Address {
-    // TODO After parsing addresses resize the memory allocation.
-    var addresses = try allocator.alloc(std.net.Address, address_limit);
+    const address_count = std.mem.count(u8, raw, ",") + 1;
```

Comment:
We already count the commas at [c/tb_client/thread.zig](https://github.com/tigerbeetledb/tigerbeetle/blob/a15fa7523385b826d85c6294f374314369f3913a/src/c/tb_client/thread.zig#L101). Would it be better to remove the `address_limit` parameter and use `config.replicas_max` instead?

Reply:
Thanks @batiati 

I think that `c/tb_client/thread.zig` shouldn't count, and delegate this rather to `parse_addresses()`, passing `config.replicas_max` as `address_limit`. `address_limit` is useful as a parameter here to decouple from `config`, i.e. for testing.